### ISSUE TITLE
Download file and it's metadata with one call

### ISSUE THREAD

**Issue by casamia918** [strongloop/loopback-component-storage#124]
I know that loopback-storage-component is not yet provide metadata saving while upload a file. So someone solved this problem by creating a metadata model which 1 to 1 relationship with file. 
http://stackoverflow.com/questions/28885282/how-to-store-files-with-meta-data-in-loopback

He solved the downloading problem by saving the download url in the metadata model. So, when client want to download some file, first it request the metadata object using metadata id (GET /MetaFile/:id), second fetch the download url, and last re-request the download url(GET /Containers/:container/File/:file). 

Summing up, if client need a file from server, there must be a 2 API request, which is doubled and seems to abnormal.

So, I want to make a 1 request way, which is resembled to the upload. When user request download (GET /MetaFile/:id/download), user get metadata And the filestream of the related file. Of course, the filestream is got from the server. 

My question is, how to put in the metadata (which is json object) and filestream to the return message at one time. I think it's not impossible but it looks likes needs some knowledge.
